docker/app
# WIP: refactor as a CLI plugin

**- What I did**

First pass refactoring as a plugin using the framework from https://github.com/docker/cli/pull/1564.

I intend to track that PR using app as a test case for the framework. Shouldn't be merged until the plugins stuff is available!

**- How to verify it**

With a `docker` CLI binary build from https://github.com/docker/cli/pull/1564 place the `docker-app` built from this PR in the plugin search path (or set `DOCKER_CLI_PLUGIN_EXTRA_DIRS` to point to a directory containing `docker-app`). Then `docker app «foof»` should work as `docker-app «foof»` did previously.
